source/common/router/router_filter_interface.h

```diff
@@ -0,0 +1,58 @@
+#pragma once
+
+#include "envoy/http/filter.h"
+#include "envoy/http/header_map.h"
+#include "envoy/stats/scope.h"
+#include "envoy/stats/stats_macros.h"
+#include "envoy/stream_info/stream_info.h"
+#include "envoy/upstream/cluster_manager.h"
+#include "envoy/upstream/host_description.h"
+
+namespace Envoy {
+namespace Router {
+
+class UpstreamRequest;
+
+struct TimeoutData {
+  std::chrono::milliseconds global_timeout_{0};
+  std::chrono::milliseconds per_try_timeout_{0};
+  std::chrono::milliseconds per_try_idle_timeout_{0};
+};
+
+// The interface the UpstreamRequest has to interact with the router filter.
+class RouterFilterInterface {
+public:
+  virtual ~RouterFilterInterface() = default;
+
+  virtual void onUpstream1xxHeaders(Http::ResponseHeaderMapPtr&& headers,
```

Comment:
Can we add brief comments for these methods? Their names are pretty readable, but a comment is always helpful.